core/src/main/java/hudson/Functions.java

```diff
@@ -201,7 +202,25 @@ public static String xsDate(Calendar cal) {
     public static String rfc822Date(Calendar cal) {
         return Util.RFC822_DATETIME_FORMATTER.format(cal.getTime());
     }
-    
+
+    /**
+     * During Jenkins start-up, before {@link InitMilestone#PLUGINS_STARTED} the extensions lists will be empty
+     * and they are not guaranteed to be fully populated until after {@link InitMilestone#EXTENSIONS_AUGMENTED}.
+     * If you attempt to access the extensions list from a UI thread while the extensions are being loaded you will
+     * hit a big honking great monitor lock that will block until the effective extension list has been determined
+     * (as if a plugin fails to start, all of the failed plugin's extensions and any dependent plugins' extensions
+     * will have to be evicted from the list of extensions. In practical terms this only affects the
+     * "Jenkins is loading" screen, but as that screen uses the generic layouts we provide this utility method
+     * so that the generic layouts can avoid iterating extension lists while Jenkins is starting up.
+     *
+     * @return {@code true} if the extensions lists have been populated.
+     * @since 1.FIXME
+     */
+    public static boolean isExtensionsAvailable() {
```

Comment:
The java beans specification states that getters start with `get` unless the return type is `boolean` in which case they should start with `is`. As this method is accessed via EL it has to be named in accordance with the Java Beans specification, i.e. `is` that way in the jelly page we can do `<j:if test="${h.extensionsAvailable}">...</j:if>`